tests/Unit/ApparentHorizons/Test_FastFlow.cpp

```diff
@@ -0,0 +1,283 @@
+                                                 200));
+}
+
+void test_construct_from_options_curvature() {
+  Options<tmpl::list<FastFlowFromOpts>> opts("");
+  opts.parse(
+      "FastFlowFromOpts:\n"
+      "  Flow: Curvature\n"
+      "  Alpha: 1.1\n"
+      "  Beta: 0.6\n"
+      "  AbsTol: 1.e-10\n"
+      "  TruncationTol: 1.e-3\n"
+      "  DivergenceTol: 1.1\n"
+      "  DivergenceIter: 6\n"
+      "  MaxIts: 200");
+  CHECK(opts.get<FastFlowFromOpts>() == FastFlow(FastFlow::FlowType::Curvature,
+                                                 1.1, 0.6, 1e-10, 1e-3, 1.1, 6,
+                                                 200));
+}
+
+void test_serialize() noexcept {
+  FastFlow fastflow(FastFlow::FlowType::Jacoby, 1.1, 0.6, 1e-10, 1e-3, 1.1, 6,
+                    200);
+  test_serialization(fastflow);
+}
+
+void test_copy_and_move() noexcept {
+  FastFlow fastflow(FastFlow::FlowType::Curvature, 1.1, 0.6, 1e-10, 1e-3, 1.1,
+                    6, 200);
+  test_copy_semantics(fastflow);
+  auto fastflow_copy = fastflow;
+  // clang-tidy: std::move of triviable-copyable type
+  test_move_semantics(std::move(fastflow), fastflow_copy);  // NOLINT
+}
+
+void test_ostream() noexcept {
+  {
+    std::ostringstream stream;
+    stream << FastFlow::FlowType::Curvature;
+    CHECK(stream.str() == "Curvature");
+  }
+  {
+    std::ostringstream stream;
+    stream << FastFlow::FlowType::Jacoby;
+    CHECK(stream.str() == "Jacoby");
+  }
+  {
+    std::ostringstream stream;
+    stream << FastFlow::FlowType::Fast;
+    CHECK(stream.str() == "Fast");
+  }
+}
+
+void test_negative_radius_error() {
+  // Set initial Strahlkorper radius to negative on purpose to get
+  // error exit status.
+  Strahlkorper<Frame::Inertial> strahlkorper(5, 5, -1.0, {{0, 0, 0}});
+  FastFlow flow(FastFlow::FlowType::Fast, 1.0, 0.5, 1e-12, 1e-10, 1.2, 5, 100);
+
+  EinsteinSolutions::KerrSchild solution(1.0, {{0., 0., 0.}}, {{0., 0., 0.}});
+
+  auto status = do_iteration(&strahlkorper, &flow, solution);
+  CHECK(status == FastFlow::Status::NegativeRadius);
+}
+
+void test_too_many_iterations_error() {
+  Strahlkorper<Frame::Inertial> strahlkorper(5, 5, 3.0, {{0, 0, 0}});
+  // Set number of iterations to 1 on purpose to get error exit status.
+  FastFlow flow(FastFlow::FlowType::Fast, 1.0, 0.5, 1e-12, 1e-10, 1.2, 5, 1);
+
+  EinsteinSolutions::KerrSchild solution(1.0, {{0., 0., 0.}}, {{0., 0., 0.}});
+
+  auto status = do_iteration(&strahlkorper, &flow, solution);
+  CHECK(status == FastFlow::Status::MaxIts);
+}
+
+void test_schwarzschild(FastFlow::Flow::type type_of_flow,
+                        const size_t max_iterations) {
+  Strahlkorper<Frame::Inertial> strahlkorper(5, 5, 3.0, {{0, 0, 0}});
+  FastFlow flow(type_of_flow, 1.0, 0.5, 1e-12, 1e-10, 1.2, 5, max_iterations);
+
+  EinsteinSolutions::KerrSchild solution(1.0, {{0., 0., 0.}}, {{0., 0., 0.}});
+
+  auto status = do_iteration(&strahlkorper, &flow, solution);
+  CHECK(static_cast<int>(status) > 0);
```

Comment:
write a free function that returns
`bool succeeded(decltype(status))` so we don't rely on the values of the enum outside FastFlow